config/config_test.go

```diff
@@ -0,0 +1,110 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestConfigParser(t *testing.T) {
+	Convey("Parser should accept valid string", t, func() {
+		cfg, err := ParseConfig(configString)
+		So(cfg, ShouldNotBeNil)
+		So(err, ShouldBeNil)
+		validateConfig(cfg)
+	})
+
+	Convey("Parser should reject invalid strings - no listener config", t, func() {
+		cfg, err := ParseConfig(`
+			# no listener config
+			vault {
+				address       = "http://127.0.0.8200"
+			}
+			`)
+		So(cfg, ShouldBeNil)
+		So(err, ShouldNotBeNil)
+	})
+
+	Convey("Parser should reject invalid strings - no vault config", t, func() {
+		cfg, err := ParseConfig(`
+			listener "tcp" {
+				address       = "127.0.0.1:8000"
+			}
+			# no vault config
+			`)
+		So(cfg, ShouldBeNil)
+		So(err, ShouldNotBeNil)
+	})
+
+	Convey("Starting up a dev vault", t, func() {
+		cfg, shutdownCh, secretID, err := LoadConfigDev()
+		validateConfig(cfg)
+		So(shutdownCh, ShouldNotBeNil)
+		defer close(shutdownCh)
+		So(secretID, ShouldNotBeNil)
+		So(err, ShouldBeNil)
+		validateVaultHealth(cfg)
+	})
+
+	Convey("Loading custom config", t, func() {
+		cfg, err := LoadConfigFile("sample.hcl")
```

Comment:
Looks like it. This seems concerning as it will become tricky to develop tests if it runs different on the CI. Any idea how we can achieve consistent results between the two environments?